cate/core/opimpl.py

```diff
@@ -650,12 +654,19 @@ def _get_geo_spatial_cf_attrs_from_var(ds: xr.Dataset, var_name: str, allow_poin
                 dim_res = abs(var.values[1] - var.values[0])
                 dim_min = min(var.values[0], var.values[-1]) - 0.5 * dim_res
                 dim_max = max(var.values[0], var.values[-1]) + 0.5 * dim_res
-            elif len(var.values) == 1 and allow_point:
-                dim_var = var
-                # Actually a point with no extent
-                dim_res = 0.0
-                dim_min = var.values[0]
-                dim_max = var.values[0]
+            elif len(var.values) == 1:
+                if res_name in ds.attrs:
+                    dim_var = var
+                    # Consider extent in metadata if provided
+                    dim_res = float(ds.attrs[res_name].split(' ')[0])
```

Comment:
To what does this magic refer to: `ds.attrs[res_name].split(' ')[0]`

Reply:
If you know it, say so why you split at " ", ignoring everything in the result but the first part.

Reply:
Right, I'll ,make this clearer. It is not correct as it is anyway.